airbyte-integrations/connectors/source-square/source_square/source.py

```diff
@@ -0,0 +1,421 @@
+
+        json_payload.update({"limit": self.items_per_page_limit})
+        return json_payload
+
+
+class TeamMemberWages(SquareStream):
+    """ Docs: https://developer.squareup.com/reference/square_2021-06-16/labor-api/list-team-member-wages """
+
+    data_field = "team_member_wages"
+    items_per_page_limit = 200
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "labor/team-member-wages"
+
+    def request_params(self, **kwargs) -> MutableMapping[str, Any]:
+        params_payload = super().request_params(**kwargs)
+        if not params_payload:
+            params_payload = {}
+
+        if "next_page_token" in kwargs and kwargs["next_page_token"]:
+            params_payload.update({"cursor": kwargs["next_page_token"]["cursor"]})
+
+        params_payload.update({"limit": self.items_per_page_limit})
+        return params_payload
+
+
+class Customers(SquareStream):
+    """ Docs: https://developer.squareup.com/reference/square_2021-06-16/customers-api/list-customers """
+
+    data_field = "customers"
+
+    def path(self, **kwargs) -> str:
+        return "customers"
+
+    def request_params(self, **kwargs) -> MutableMapping[str, Any]:
+        params_payload = super().request_params(**kwargs)
+        if not params_payload:
+            params_payload = {}
+
+        if "next_page_token" in kwargs and kwargs["next_page_token"]:
+            params_payload.update({"cursor": kwargs["next_page_token"]["cursor"]})
+
+        params_payload.update({"sort_order": "ASC", "sort_field": "CREATED_AT"})
+        return params_payload
+
+
+class Orders(SquareStream):
+    """ Docs: https://developer.squareup.com/reference/square/orders-api/search-orders """
+
+    data_field = "orders"
+    http_method = "POST"
+    items_per_page_limit = 500
+
+    def path(self, **kwargs) -> str:
+        return "orders/search"
+
+    def request_body_json(self, **kwargs) -> Optional[Mapping]:
+        json_payload = super().request_body_json(**kwargs)
+        if not json_payload:
+            json_payload = {}
+
+        args = {
+            "authenticator": self.authenticator,
+            "is_sandbox": self.is_sandbox,
+            "api_version": self.api_version,
+            "start_date": self.start_date,
+            "include_deleted_objects": self.include_deleted_objects,
+        }
+
+        location_ids = []
+        for location_item in Locations(**args).read_records(sync_mode=SyncMode.full_refresh):
+            location_ids.append(location_item["id"])
+
+        if location_ids:
+            json_payload.update({"location_ids": location_ids})
+
+        if "next_page_token" in kwargs and kwargs["next_page_token"]:
+            json_payload.update({"cursor": kwargs["next_page_token"]["cursor"]})
+
+        json_payload.update({"limit": self.items_per_page_limit})
+        return json_payload
```

Comment:
does api returns successful response if there no locations ?

Reply:
No, it fails with a 400 Response error.
Do you suggest returning 0 records in this case?

Reply:
yes, I think it's better to skip it if we know that request is gonna fail

Reply:
Done. Return 0 records if locations are empty